src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/show_request.cc

```diff
@@ -94,6 +99,125 @@ void ShowRequest::HandleMetadata(const char* field_id,
   }
 }
 
+bool ShowRequest::CheckTemplateConfigurationForApp(
+    application_manager::Application& app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if ((*message_)[strings::msg_params].keyExists(strings::window_id)) {
+    current_window_id_ =
+        (*message_)[strings::msg_params][strings::window_id].asInt();
+  }
+
+  const auto set_window_layout = [&app, this]() -> bool {
+    const auto new_template_layout =
+        template_config_[strings::template_layout].asString();
+    const auto old_template_layout = app.window_layout(current_window_id_);
+    LOG4CXX_DEBUG(logger_, "New layout: " << new_template_layout);
+    LOG4CXX_DEBUG(logger_, "Old layout: " << old_template_layout);
+
+    const bool layouts_equal = (new_template_layout == old_template_layout);
+
+    if (!new_template_layout.empty() && !layouts_equal) {
+      // Template switched, hence allow any color change
+      LOG4CXX_DEBUG(logger_,
+                    "Show Request: Setting new Layout: " << new_template_layout
+                                                         << " for window ID: "
+                                                         << current_window_id_);
+      layout_change_required_ = true;
+      return true;
+    }
+    LOG4CXX_DEBUG(logger_, "Show Request: No Layout Change");
+    return false;
+  };
+
+  const auto set_day_color_scheme = [&app, this]() -> bool {
```

Comment:
I think set_day_color_scheme and set_night_color_scheme need to take into account if layout_change_required is true. If there is a color change requested when the template changes, and a color was previously set, it doesnt look like dcs_change_required_/ncs_change_required_ will ever be set to true.

Reply:
As of this implementation,`Show` request duplicates logic of `SetDisplayLayout` request. As stated in [Color Scheme](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0147-template-color-scheme.md) proposal, 
`SDL Core should track the number of attempted SetDisplayLayout requests with the current template and REJECT any beyond the first with the reason "Using SetDisplayLayout to change the color scheme may only be done once.`

Reply:
The key wording here is:

> with the current template

The developer should be able to change the color scheme after the initial color scheme any time the template layout is changed.

Reply:
Here is the existing behavior in set display layout request. I believe this behavior should be transferred to the show requests logic handling of color schemes.

https://github.com/smartdevicelink/sdl_core/blob/master/src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_display_layout_request.cc#L78-L82

Reply:
Also referencing the comment in your code,

> // Template switched, hence allow any color change

Reply:
@JackLivio fixed in ebb5dcc5575fe835d3b64522cd8015a00af55cc9